src/App.js

```diff
@@ -1,16 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likesCount, setLikesCount] = useState(0);
+
+  const increaseLikes = () => {
+    setLikesCount(likesCount + 1);
+  };
```

Comment:
The project specs were a little unclear as to what the total liked count was supposed to track. The intention was to give a count of the currently liked messages in the conversation (so if 3 messages were currently liked, we'd see 3, but if we then click one so that it's no longer liked, the count should decrease to 2. With that behavior, of we store the list of messages in state here, we would be able to calculate the current number of liked messages using that state data, and would be able to eliminate the separate piece of state and logic tracking the count of likes!

However, with the interpretation you went with (which I can understand and which passes the tests), you're tallying up how many times something was liked. In that case, we _do_ need an additional piece of state, since the current list of messages wouldn't not contain the historical data we would need.

I'd would suggest revisiting the behavior here to go more for the "currently liked count" behavior over the "historically liked count" behavior, as that would give you a chance to think about removing this additional state. But, even if we stick with the "historical" approach, then rather than having the ChatEntry component need to make a decision about whether to notify that the like count needs to change, we should bring that logic up into an id-based callback here.

The ChatEntry could call a function taking the id of the message that was clicked. Our logic up here (living near to where the state is actually tracked) would update the messages state by toggling the liked status of the matching message. _But it would also know whether that liked status was being toggled to `true`!_ So the logic here would also be able to update the like count appropriately.